### PR TITLE
feat: integrate RPAv3 opt from tpu inference

### PR DESCRIPTION
**Pull request by Iamleos**
## Accuracy Tests
image

#### server command:
`JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path /models/models--Qwen--Qwen3-8B --trust-remote-code --device=tpu --random-seed=3 --mem-fraction-static=0.8 --chunked-prefill-size=4096 --download-dir=/dev/shm --dtype=bfloat16 --max-running-requests 128 --skip-server-warmup --page-size=128 --disable-radix-cache`
#### bench command:
`python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 128 --random-input xxxx --random-output 1024 --random-range-ratio 1 --warmup-requests 0`
|                        | Median TTFT(ms) | Median ITL(ms) | Output Throughput |
| ------------- | ------------- | ----------- | -------------------- |
| in1k_out_1k| 23001.21 | 19.9 | 1530.8|
| in4k_out1k | 96212.8 (3% ⬆️) | 21.03 (2.8%⬆️) | 559.63 (3%⬆️)|

#### server command:
`JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path /models/models--Qwen--Qwen3-32B --trust-remote-code --device=tpu --tp-size=4 --random-seed=3 --mem-fraction-static=0.8 --chunked-prefill-size=4096 --download-dir=/dev/shm --dtype=bfloat16 --max-running-requests 128 --skip-server-warmup --page-size=128 --disable-radix-cache`
#### bench command:
`python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 128 --random-input xxxx --random-output 1024 --random-range-ratio 1 --warmup-requests 0`
|                        | Median TTFT(ms) | Median ITL(ms) | Output Throughput |
| ------------- | ------------- | ----------- | -------------------- |
| in1k_out_1k| 3204.7 (2.9% ⬆️) | 20.75 (13.8% ⬆️) | 2693.37 (5.6% ⬆️)|
| in4k_out1k | 60930.39 | 23.23 | 953.5|

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

**Comment by JamesBrianD** [comment]
@depksingh Could you help review this PR?

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py [hunk 1]
```diff
@@ -21,664 +21,1184 @@
...
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 16): (32, 2),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 32): (32, 4),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 64): (32, 8),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 128): (32, 16),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 256): (32, 2),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 512): (16, 128),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 1024): (16, 128),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 2048): (32, 128),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 4096): (32, 128),
-        ("bfloat16", "bfloat16", 2, 2, 128, 64, 8192): (32, 128),
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:37]: It appears that there is no tune configuration with page-size = 64.

Comment (Iamleos) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:37]: Yes, I will do it later

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py [hunk 2]
```diff
@@ -21,664 +21,1184 @@
...
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 64): (2, 1),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 128): (2, 1),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 256): (2, 1),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 512): (2, 128),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 1024): (4, 128),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 2048): (4, 128),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 4096): (4, 128),
+        ("bfloat16", "bfloat16", 32, 32, 128, 256, 8192): (4, 128),
+    },
+    "TPU v7": {
```
Comment (depksingh) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:611]: I think this should be "TPU7x", Can you please check

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py [hunk 3]
```diff
@@ -692,7 +1212,7 @@ def get_tuned_block_sizes(
     page_size,
     max_num_tokens,
     pages_per_seq,
-    causal,
+    causal=True,
```
Comment (depksingh) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:1213]: Can we please add a log line on else block of line 1245, which can give user warning that currentlly the config being used isn't having a tuned size, and also if the code isn't picking up the tuned number, the user might know, else this might get unnoticed.

```
else:
print(
    f"Tuned RPA block sizes not found for {device_name}: {page_size=}, {actual_num_q_heads=}, {actual_num_kv_heads=}, {head_dim=}, {max_num_tokens=}, {pages_per_seq=}. Using default block sizes."
)
```

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py [hunk 4]
```diff
@@ -21,664 +21,1184 @@
...
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 32): (16, 1),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 64): (32, 1),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 128): (16, 1),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 256): (16, 1),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 512): (4, 128),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 1024): (8, 128),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 2048): (16, 128),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 4096): (16, 128),
+        ("bfloat16", "bfloat16", 1, 1, 128, 128, 8192): (16, 128),
+        ("bfloat16", "bfloat16", 1, 1, 128, 256, 1): (8, 1),
```
Comment (depksingh) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:36]: Are these numbers deterministic now? Like last time I tried rerunning the tuning for the same configuration b2b and it gave different results for the same run, just a sanity check before updating these numbers.

File: python/sgl_jax/srt/kernels/utils/perf.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    task: str,
+    tries: int = 5,
+    trace_root: str = "/tmp/sglang_jax_moe_trace",
+) -> list[float]:
+    """
+    Profile multiple iterations and pull per-iteration kernel time from trace.
+    """
+    trace_name = f"{task}_" + "".join(random.choices(string.ascii_uppercase + string.digits, k=6))
+    trace_dir = os.path.join(trace_root, trace_name)
+    os.makedirs(trace_dir, exist_ok=True)
```
Comment (depksingh) [inline python/sgl_jax/srt/kernels/utils/perf.py:85]: Shall we delete the dir after this trace analysis is done?

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py (hunk 2 above)
Comment (JamesBrianD) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:611]: image

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py (hunk 3 above)
Comment (Iamleos) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:1213]: Good suggestion

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py (hunk 4 above)
Comment (Iamleos) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:36]: We use tracing file to get the kernel time. I have already checked. The numbers is deterministic with the same input ans config.

File: python/sgl_jax/srt/kernels/utils/perf.py (hunk 5 above)
Comment (Iamleos) [reply python/sgl_jax/srt/kernels/utils/perf.py:85]: I think it is unnecessary. Perf code is only used in tested environment.

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py [hunk 6]
```diff
@@ -2,11 +2,9 @@
 
 import jax.numpy as jnp
 
-from sgl_jax.srt.kernels.ragged_paged_attention.util import (
-    get_device_name,
-    get_tpu_version,
-)
+from sgl_jax.srt.kernels.ragged_paged_attention.util import get_tpu_version
 from sgl_jax.srt.utils.common_utils import next_power_of_2
+from sgl_jax.srt.utils.jax_utils import get_device_name
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:7]: `get_device_name` in sgl_jax.srt.kernels.ragged_paged_attention.util is unused. Could it be deleted.

**Review by JamesBrianD — APPROVED** [review]

File: python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py (hunk 1 above)
Comment (depksingh) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/tuned_block_sizes.py:37]: @Iamleos , @JamesBrianD , as the script is updated now, I can run it for pg_sz 64, if you haven't already, Please let me know.
